library/aes.c

```diff
@@ -323,31 +337,44 @@ static const uint32_t RCON[10] =
     0x00000010, 0x00000020, 0x00000040, 0x00000080,
     0x0000001B, 0x00000036
 };
+#endif /* !defined(MBEDTLS_AES_SETKEY_ENC_ALT) */
 
 #else /* MBEDTLS_AES_ROM_TABLES */
 
 /*
  * Forward S-box & tables
  */
+#if !defined(MBEDTLS_AES_ENCRYPT_ALT) || !defined(MBEDTLS_AES_SETKEY_ENC_ALT) || \
+    !defined(MBEDTLS_AES_SETKEY_DEC_ALT)
 static unsigned char FSb[256];
+#endif /* !defined(MBEDTLS_AES_ENCRYPT_ALT) || !defined(MBEDTLS_AES_SETKEY_ENC_ALT) || \
```

Comment:
Actually, the `#endif` is so close to the `#if` that it's more readable without the comment on the end (but with a blank line before the next `#if` at line 352)

Reply:
There are so many #endif's in this file I think being explicitly unambiguous may be the better choice here. I will leave this unless other reviewers feel the same (same for similar comments below)

Reply:
Of course.  My position remains that, precisely because of so many `#if`s, the comments on `#endif`s that are so close to their `#if`s mean the actual code that is being guarded is drowned out by the comment, and readability is much enhanced without them